packages/xstate-graph/src/types.ts

```diff
@@ -173,9 +176,9 @@ export interface TraversalOptions<TState, TEvent extends EventObject>
 }
 
 export type EventCaseMap<TState, TEvent extends EventObject> = {
-  [TEventType in TEvent['type']]?:
-    | ((state: TState) => Array<EventCase<TEvent>>)
-    | Array<EventCase<TEvent>>;
+  [E in TEvent as E['type']]?:
```

Comment:
It requires TS 4.1 but I feel that for `@xstate/graph`/`@xstate/test` it's already OK - this version correctly "narrows down" the `TEvent` type per the property key in this mapped type